src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -10163,6 +10163,108 @@ def resample(
 
         return SnowflakeQueryCompiler(frame)
 
+    def value_counts_index(
+        self,
+        normalize: bool = False,
+        sort: bool = True,
+        ascending: bool = False,
+        bins: Optional[int] = None,
+        dropna: bool = True,
+    ) -> "SnowflakeQueryCompiler":
+        """
+        Counts the frequency or number of unique values of Index SnowflakeQueryCompiler.
+
+        The resulting object will be in descending order so that the
+        first element is the most frequently occurring element.
+        Excludes NA values by default.
+
+        Args:
+            normalize : bool, default False
+                If True then the object returned will contain the relative
+                frequencies of the unique values.
+            sort : bool, default True
+                Sort by frequencies when True. Preserve the order of the data when False.
+            ascending : bool, default False
+                Sort in ascending order.
+            bins : int, optional
+                Rather than count values, group them into half-open bins,
+                a convenience for ``pd.cut``, only works with numeric data.
+                This argument is not supported yet.
+            dropna : bool, default True
+                Don't include counts of NaN.
+        """
+        # TODO: SNOW-924742 Support bins in Series.value_counts
+        if bins is not None:
+            raise ErrorMessage.not_implemented("bins argument is not yet supported")
+
+        if self.is_multiindex():
+            ErrorMessage.not_implemented(
+                "`value_counts` is not yet implemented for MultiIndex objects"
+            )
+        by = self._modin_frame.index_column_pandas_labels
+
+        # validate whether by is valid (e.g., contains duplicates or non-existing labels)
+        self.validate_groupby(by=by, axis=0, level=None)
+
+        # append a dummy column for count aggregation
+        COUNT_LABEL = "value_count"
+        query_compiler = SnowflakeQueryCompiler(
```

Comment:
Here we seems duplicating the code in snowflake_query_compiler.value_counts, which is not a good practice, can you please extract the common code out to a common function used by both interface, and calls into it accordingly?